hazelcast-client/src/main/java/com/hazelcast/client/impl/querycache/subscriber/QueryCacheToListenerMapper.java

```diff
@@ -49,31 +49,45 @@
         this.registrations = new ConcurrentHashMap<String, Collection<ListenerInfo>>();
     }
 
-    public String addListener(String cacheName, ListenerAdapter listenerAdapter, EventFilter filter) {
-        Collection<ListenerInfo> adapters = getOrPutIfAbsent(registrations, cacheName, LISTENER_SET_CONSTRUCTOR);
+    public String addListener(String cacheId, ListenerAdapter listenerAdapter, EventFilter filter) {
+        Collection<ListenerInfo> adapters = getOrPutIfAbsent(registrations, cacheId, LISTENER_SET_CONSTRUCTOR);
         String id = UuidUtil.newUnsecureUuidString();
         ListenerInfo info = new ListenerInfo(filter, listenerAdapter, id);
         adapters.add(info);
         return id;
     }
 
-    public boolean removeListener(String cacheName, String id) {
-        Collection<ListenerInfo> adapters = getOrPutIfAbsent(registrations, cacheName, LISTENER_SET_CONSTRUCTOR);
+    public boolean removeListener(String cacheId, String listenerId) {
+        Collection<ListenerInfo> adapters = getOrPutIfAbsent(registrations, cacheId, LISTENER_SET_CONSTRUCTOR);
         Iterator<ListenerInfo> iterator = adapters.iterator();
         while (iterator.hasNext()) {
             ListenerInfo listenerInfo = iterator.next();
             String listenerInfoId = listenerInfo.getId();
-            if (listenerInfoId.equals(id)) {
+            if (listenerInfoId.equals(listenerId)) {
                 iterator.remove();
                 return true;
             }
         }
         return false;
     }
 
+    /**
+     * Removes all associated listener info for the cache represented with the supplied {@code cacheId}
+     *
+     * @param cacheId id of the cache
+     */
+    public void removeAllListeners(String cacheId) {
+        registrations.remove(cacheId);
+    }
+
+    // this method is only used for testing purposes
```

Comment:
Looks like this method is also used in production code, e.g. `ClientQueryCacheEventService`?

Reply:
very good catch. fixed.